File: Exercise/Hard/05. Password Strength Validator/Code/Password Strength Validator.cs
using System;

class Program {
    static void Main() {
        string password = Console.ReadLine();

        if (IsStrongPassword(password)) {
            Console.WriteLine("Strong");
        }
        else {
            Console.WriteLine("Weak");
        }
    }

    static bool IsStrongPassword(string password) {
        bool hasLength = password.Length >= 8;
        bool hasUpperCase = false;
        bool hasLowerCase = false;
        bool hasDigit = false;
        bool hasSpecialChar = false;

        foreach (char ch in password) {
            if (char.IsUpper(ch)) {
                hasUpperCase = true;
            }
            else if (char.IsLower(ch)) {
                hasLowerCase = true;
            }
            else if (char.IsDigit(ch)) {
                hasDigit = true;
            }
            else if (char.IsPunctuation(ch) || char.IsSymbol(ch)) {
                hasSpecialChar = true;
            }
        }

        return hasLength && hasUpperCase && hasLowerCase && hasDigit && hasSpecialChar;
    }
}

File: Exercise/Hard/05. Password Strength Validator/Code/Password Strength Validator.py
def is_strong_password(password):
    has_length = len(password) >= 8
    has_uppercase = any(char.isupper() for char in password)
    has_lowercase = any(char.islower() for char in password)
    has_digit = any(char.isdigit() for char in password)
    has_special_char = any(char in "!@#$%^&*()-_+=<>?" for char in password)

    return has_length and has_uppercase and has_lowercase and has_digit and has_special_char

def main():
    password = input().strip()

    if is_strong_password(password):
        print("Strong")
    else:
        print("Weak")

if __name__ == "__main__":
    main()
